qmi/core/transport.py

```diff
@@ -686,22 +689,34 @@ def read_until(self, message_terminator: bytes, timeout: Optional[float]) -> byt
                 tremain = tstart + timeout - time.monotonic()
                 if tremain < 0:
                     nbuf = len(self._read_buffer)
-                    raise QMI_TimeoutException("Timeout after {} bytes without message terminator".format(nbuf))
+                    raise QMI_TimeoutException(f"Function timeout after {nbuf} bytes without message terminator")
                 self._safe_socket.settimeout(tremain)
+
             # Read from socket.
             try:
-                # NOTE: Reading up to 4096 bytes here.
+                # NOTE: Reading up to 512 bytes here for TCP, as larger packet size
```

Comment:
should we really have this limit for only one instrument? Can we have an option in `read_until` to specify how many bytes we read?

Reply:
For the moment this is necessary to do, as otherwise the Tektronix AWG won't work properly (and possibly same issue could be with other instruments using VXI11 transport, but not seen so far). I would not want to change here the packet size, and changing the method signature is also not possible (Mypy...) as we inherit this from `QMI_Transport` and we should use the same signature.

There is the ticket on our board with name "Rename QMI_Transport read call names and modify their inputs for more logical naming" where we could re-open it for discussion, to see if we could include this. It would help in this case.

Reply:
sounds good! in that ticket can we link this comment/issue so we don't forget

Reply:
I commented that at the start of the issue and will refine the ticket later on.